c/python/tuple.cc

```diff
@@ -20,10 +20,18 @@
 // IN THE SOFTWARE.
 //------------------------------------------------------------------------------
 #include "python/tuple.h"
+#include "utils/assert.h"
 
 namespace py {
 
 
+//------------------------------------------------------------------------------
+// Destructor
+//------------------------------------------------------------------------------
+
+otuple::~otuple() {}
```

Comment:
tuple doesn't become a virtual class, it only gets one virtual function that is defined in the same class. Actually, I was not even going to use virtual functions, but I was like this is what you suggested when we discussed that yesterday?

Reply:
A class becomes virtual whenever it gets even a single virtual method.
I agree that this is what we discussed yesterday, but I didn't realize then that it would force the class to become virtual. Still, it's quite simple to avoid virtual functions here: see my suggestion in a comment below.

Reply:
Sorry, I confused it with the abstract class. There is no problem for me to avoid virtual classes/functions.